protocol-units/da/m1/light-node-verifier/src/lib.rs

```diff
@@ -1,44 +1,44 @@
-pub mod v1;
+pub mod celestia;
+pub mod permissioned_signers;
+pub mod signed;
 
 pub use m1_da_light_node_grpc::*;
+use thiserror::Error;
 
-#[tonic::async_trait]
-pub trait Verifier {
-	async fn verify(
-		&self,
-		verification_mode: VerificationMode,
-		blob: &[u8],
-		height: u64,
-	) -> Result<bool, anyhow::Error> {
-		match verification_mode {
-			VerificationMode::Cowboy => self.verify_cowboy(verification_mode, blob, height).await,
-			VerificationMode::ValidatorIn => {
-				self.verifiy_validator_in(verification_mode, blob, height).await
-			}
-			VerificationMode::MOfN => self.verify_m_of_n(verification_mode, blob, height).await,
-		}
+/// Domain error for the transaction pipe task
+#[derive(Debug, Error)]
+pub enum Error {
+	#[error("verifier internal error: {0}")]
+	Internal(String),
+	#[error("verifier validation error: {0}")]
+	Validation(String),
+}
+
+/// thiserror for validation and internal errors
+#[derive(thiserror::Error, Debug)]
+
+/// A verified outcome. Indicates that input of A (from the trait [VerifierOperations]) is verified as valid instance of B, or else invalid instance.
+pub struct Verified<B>(B);
+
+impl<B> Verified<B> {
+	pub fn new(blob: B) -> Self {
+		Self(blob)
 	}
 
-	async fn verify_cowboy(
-		&self,
-		_verification_mode: VerificationMode,
-		_blob: &[u8],
-		_height: u64,
-	) -> Result<bool, anyhow::Error> {
-		Ok(true)
+	pub fn inner(&self) -> &B {
+		&self.0
 	}
 
-	async fn verifiy_validator_in(
-		&self,
-		_verification_mode: VerificationMode,
-		_blob: &[u8],
-		_height: u64,
-	) -> Result<bool, anyhow::Error>;
-
-	async fn verify_m_of_n(
-		&self,
-		_verification_mode: VerificationMode,
-		_blob: &[u8],
-		_height: u64,
-	) -> Result<bool, anyhow::Error>;
+	pub fn into_inner(self) -> B {
+		self.0
+	}
+}
+
+#[tonic::async_trait]
+pub trait VerifierOperations<A, B>
+where
+	A: Send + Sync + 'static,
+	B: Send + Sync + 'static,
```

Comment:
The `Sync` is probably falling out of some structuring of async methods and is not really required, but no need to obsess about it.